source/schema/avro.go

```diff
@@ -0,0 +1,161 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"cmp"
+	"fmt"
+	"math"
+	"slices"
+
+	"github.com/hamba/avro/v2"
+	"github.com/jackc/pglogrepl"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+const (
+	avroNS = "conduit.postgres"
+)
+
+var Avro = &avroExtractor{
+	pgMap: pgtype.NewMap(),
+	avroMap: map[string]*avro.PrimitiveSchema{
+		"bool":    avro.NewPrimitiveSchema(avro.Boolean, nil),
+		"bytea":   avro.NewPrimitiveSchema(avro.Bytes, nil),
+		"float4":  avro.NewPrimitiveSchema(avro.Float, nil),
+		"float8":  avro.NewPrimitiveSchema(avro.Double, nil),
+		"int8":    avro.NewPrimitiveSchema(avro.Long, nil),
+		"int4":    avro.NewPrimitiveSchema(avro.Int, nil),
+		"int2":    avro.NewPrimitiveSchema(avro.Int, nil),
+		"text":    avro.NewPrimitiveSchema(avro.String, nil),
+		"varchar": avro.NewPrimitiveSchema(avro.String, nil),
+		"timestamptz": avro.NewPrimitiveSchema(
+			avro.Long,
+			avro.NewPrimitiveLogicalSchema(avro.TimestampMicros),
+		),
```

Comment:
It doesn't look like Avro has any time type that supports time zones?

Reply:
> It doesn't look like Avro has any time type that supports time zones?

No, not at least as a primitive. It will stored as int64 and augmented with a logical primitive.

Reply:
Can we then use a custom schema to avoid losing the time zone information?

Reply:
I think it will be easier to assume this is the time is in UTC all the time and use that to calculate timezone offset? Wdyt?

Reply:
@hariso here is the spec https://avro.apache.org/docs/1.11.1/specification/#timestamp-microsecond-precision it assumes time is in UTC. I think what we need to make sure is that all times are in UTC when passed around in the connector. Same for dates.

Reply:
This is already the case as per this type formatter. https://github.com/ConduitIO/conduit-connector-postgres/blob/main/source/types/time.go#L28-L30

Reply:
Sounds good!